File: java/287_FindtheDuplicateNumber.java
class Solution {
    public int findDuplicate(int[] nums) {
    	int n = nums.length;
        boolean[] mark = new boolean[n];
        int res = 0;
        for(int i = 0; i < n; i++) {
        	if(mark[nums[i]] == true) {
                return nums[i];
        	}else {
				mark[nums[i]] = true;
			}
        }
        return -1;
    }
}

File: c++/287_FindtheDuplicateNumber.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        int a = 0, b = 0;
        while (1) {
            a = nums[a];
            b = nums[nums[b]];
            if (a == b) {
                a = 0;
                while (a != b) {
                    a = nums[a];
                    b = nums[b];
                }
                return a;
            }
        }
        return -1;
    }
};
